### PR TITLE
Categories in NavigationDrawer

### LINKED ISSUE #165: Support Categories
Since the server is able to handle categories (v2.2.0), the android client should be able to deal with this, too.
- Extend database, DTOs, etc.
- Extend user interface: thought about a swipeable sidebar on the left side like the android news app

### PR DESCRIPTION
**Pull request by korelstar**
This work in ready for merge, now. Closes #165.

- [x] introduce NavigationDrawer with hamburger icon
- [x] some code improvements/refactoring in `NotesListViewActivity`
- [x] move "settings" and "about" from menu to navigation
- [x] show "current" and "starred" in navigation
- [x] show every existent category in navigation
- [x] app is started with selection "current" (same behavior like before)
- [x] filter notes list in depdendeny of navigation selection
- [x] show notes in alphabetical order if category is selected
- [x] don't show starred notes above of not-starred notes (because there is the new selection "starred")
- [x] highlight current selection in navigation
- [x] set ActionBar title to indicate current category selection (?)
- [x] add icons for each navigation entry
- [x] show notes count for each navigation entry
- [x] load category list in background thread
- [x] refresh category list if notes list is refreshed
- [x] only show first-level categories in navigation / show notes of all sub-categories in notes list
- [x] show sub-category as header in notes list
- [x] set category category of new note to the currently selected category, when clicking on the create note button
- [x] keep current category selection when rotating device
- [x] don't show category in list of notes items, if a category is selected (?)
- [x] show sub-categories (second level) by clicking on the icon in the navigation (?)
- [x] check (and fix?) compatibility to note selection for single note widget
- [x] check (and fix?) indentation of changed files (4 spaces vs. tabs)
- [x] replace "/" by " / " in category titles everywhere in the GUI (this makes reading of sub-categories easier)
- [x] provide a drop-down list of existing categories in the change-category dialog
- [x] don't update changed-time of the note when updating the category (since it's meta-data, same behavior when starring a note)

### CODE REVIEW

**Comment by stefan-niedermann** [comment]
Awesome!

**Comment by korelstar** [comment]
I'm done. Please test this, everybody! And please give feedback to design issues.

**Comment by korelstar** [comment]
Here are some impressions (screenshots):

### Notes list (start activity)
main-activity

### Notes list with a category selected
notes-with-category

### Navigation drawer (only first-level categories)
navigation-drawer

### Navigation drawer with open category (click on the "plus folder" icon to show second-level categories for that category)
navigation-drawer-open-subcategories

**Comment by korelstar** [comment]
I've added some enhancements:
- [x] replace `/` by ` / ` in category titles everywhere in the GUI (this makes reading of sub-categories easier)
- [x] provide a drop-down list of existing categories in the change-category dialog
- [x] don't update changed-time of the note when updating the category (since it's meta-data, same behavior when starring a note)

**Comment by dan0xii** [comment]
I really like the functionality and it's testing ok so far on my phone. The one thing I was confused by was the default view being "Recent" but the starred notes not being grouped at the top. I suppose I think of starred as pinned and expect to see them first before anything else--maybe this wouldn't be classified as "Recent" items but a different view.

**Comment by korelstar** [comment]
Yes, it's a little bit different than before. But now, it's more aligned to other Nextcloud apps: Typically, you have a separate navigation item for starred objects.

But we can discuss this point further. I'm happy to read more opinions on this.

**Comment by korelstar** [comment]
@stefan-niedermann What's you opinion on this? Do you plan to review or should I merge this by myself?

**Comment by stefan-niedermann** [comment]
I planned to review this and #298 this saturday :)

As far as i can see from the screenshots, it seems to be great :) This is worth a little candy for me, i will contact you after final merge.

**Comment by stefan-niedermann** [comment]
Really awesome work :) Thank you @korelstar

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -12,7 +12,8 @@
         android:label="@string/app_name"
         android:networkSecurityConfig="@xml/network_security_config"
         android:supportsRtl="true"
-        android:theme="@style/OwnCloud">
```
Comment (stefan-niedermann) [inline app/src/main/AndroidManifest.xml]: Thanks, wanted to remove all possible Nextcloud/ownCloud-References long time ago...

**Comment by dan0xii** [comment]
Am I correct in thinking that I now can't get the same view that used to exist pre-category drawer, i.e. notes being sorted by starred and then modified?

**Comment by dan0xii** [comment]
Another comment on the "Recent" notes tab. As of v. 2.3.2 of the Nextcloud Notes app, notes are displayed by starred->modified, the way that the android app used to.

**Comment by korelstar** [comment]
Let's continue the discussion on the favorites sorting in #346 :-)

**Pull request merged**